napari/_qt/layer_controls/qt_labels_controls.py

```diff
@@ -445,29 +453,12 @@ def _on_color_mode_change(self):
                 self.colorModeComboBox.findData(self.layer.color_mode)
             )
 
-    def _on_editable_change(self):
-        """Receive layer model editable change event & enable/disable buttons."""
-        # In 3D mode, we need to disable all buttons other than picking
-        # (only picking works in 3D)
-        widget_list = [
-            'pick_button',
-            'fill_button',
-            'paint_button',
-            'erase_button',
-        ]
-        widgets_to_toggle = {
-            (2, True): widget_list,
-            (2, False): widget_list,
-            (3, True): widget_list,
-            (3, False): widget_list,
-        }
-
+    def _on_editable_or_visible_change(self):
+        """Receive layer model editable/visible change event & enable/disable buttons."""
         disable_with_opacity(
             self,
-            widgets_to_toggle[
-                (self.layer._slice_input.ndisplay, self.layer.editable)
-            ],
-            self.layer.editable,
+            self._EDIT_BUTTONS,
+            self.layer.editable and self.layer.visible,
```

Comment:
There's a bit of a double negative here (which was already present before). I would read `disable(button_list, True)` as "definitely disable these buttons" 😂, whereas I think here it means enable them. So I would maybe rename the function `set_buttons_enabled(button_list, True)`?

Reply:
Agreed on the double negative. I changed it to `set_widgets_enabled_with_opacity` and left it in the same place (`_qt.utils`) with the same arguments.